### PR TITLE
clusters: add support for configuring scaling when creating NPs via the MAPI cluster detail page

### PR DESCRIPTION
**Pull request by axbarsan**
Towards giantswarm/roadmap#341

No fancy stuff here, this allows us to set node pool scaling when creating node pools via the MAPI cluster details page.

## Preview

<details>
<summary>The form with the new option</summary>

image

</details>

### CODE REVIEW

File: src/components/shared/NodeCountSelector.js [hunk 1]
```diff
@@ -57,7 +59,7 @@ class NodeCountSelector extends React.Component {
 
     if (this.props.autoscalingEnabled === true) {
       return (
-        <form onSubmit={this.handleFormSubmit}>
```
Comment (axbarsan) [inline src/components/shared/NodeCountSelector.js:60]: This shouldn't be a `<form />` element. This is an input, it should be part of a `<form />` element together with other inputs. Using a `<form />` element inside another one is invalid HTML markup

File: src/components/shared/NodeCountSelector.js [hunk 2]
```diff
@@ -16,15 +16,17 @@ const DEFAULT_VALUE_CONSTRAINTS = {
  * max values for scaling limits.
  */
 class NodeCountSelector extends React.Component {
-  handleFormSubmit = (e) => {
-    e.preventDefault();
-  };
-
   updateValue(valuesToAdd) {
-    const nextScalingValue = Object.assign({}, this.props.scaling, valuesToAdd);
+    setTimeout(() => {
```
Comment (axbarsan) [inline src/components/shared/NodeCountSelector.js:20]: Forcing the callback to be executed after `react` makes all the updates.

This is because the 2 underlying `<NumberPicker />` components execute the callback when their `min`/`max` props change (because they also need to validate again). This also includes the first time that the `min`/`max` props are set, which is when the components get mounted.

Because the 2 callbacks (for 2 `<NumberPicker />` components) get executed one right after another, the value of the scaling in the props does not have time update in between calls. Therefore the callback returns that the input is invalid when it's actually valid.

Long story short, this makes sure that the scaling is valid by default and it doesn't prevent node pool creation.

File: src/components/shared/NodeCountSelector.js [hunk 3]
```diff
@@ -29,13 +31,22 @@ class NodeCountSelector extends React.Component {
...
+      });
+    });
   };
 
   updateScalingMax = (numberPicker) => {
-    this.updateValue({
-      max: numberPicker.value,
-      maxValid: numberPicker.valid,
+    clearTimeout(this.updateTimeout);
+    this.updateTimeout = setTimeout(() => {
```
Comment (axbarsan) [inline src/components/shared/NodeCountSelector.js:45]: Forcing the callback to be executed after `react` makes all the updates.

This is because the 2 underlying `<NumberPicker />` components execute the callback when their `min`/`max` props change (because they also need to validate again). This also includes the first time that the `min`/`max` props are set, which is when the components get mounted.

Because the 2 callbacks (for 2 `<NumberPicker />` components) get executed one right after another, the value of the scaling in the props does not have time update in between calls. Therefore the callback returns that the input is invalid when it's actually valid.

Long story short, this makes sure that the scaling is valid by default and it doesn't prevent node pool creation."

File: src/components/UI/Inputs/NumberPicker/index.tsx [hunk 4]
```diff
@@ -187,7 +187,7 @@ const NumberPicker = React.forwardRef<HTMLInputElement, INumberPickerProps>(
     };
 
     useEffect(() => {
-      if (prevMax !== max || prevMin !== min) {
+      if ((prevMax && prevMax !== max) || (prevMin && prevMin !== min)) {
```
Comment (axbarsan) [inline src/components/UI/Inputs/NumberPicker/index.tsx:190]: This is to prevent `calling the callback` when the component is rendered for the first time

**Review by marians — APPROVED** [review]
LGTM if it works for you
